File: src/main/java/com/ject/studytrip/mission/domain/model/DailyMission.java
package com.ject.studytrip.mission.domain.model;

import com.ject.studytrip.global.common.entity.BaseTimeEntity;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.*;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder(access = AccessLevel.PRIVATE)
public class DailyMission extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mission_id", nullable = false)
    private Mission mission;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "daily_goal_id", nullable = false)
    private DailyGoal dailyGoal;

    public static DailyMission of(Mission mission, DailyGoal dailyGoal) {
        return DailyMission.builder().mission(mission).dailyGoal(dailyGoal).build();
    }

    public void updateDeletedAt() {
        this.deletedAt = LocalDateTime.now();
    }

    public DailyGoal getDailyGoal() {
        return dailyGoal;
    }

    public Long getId() {
        return id;
    }

    public Mission getMission() {
        return mission;
    }
}

File: src/main/kotlin/com/ject/studytrip/mission/domain/model/DailyMission.kt
package com.ject.studytrip.mission.domain.model

import com.ject.studytrip.global.common.entity.BaseTimeEntity
import com.ject.studytrip.trip.domain.model.DailyGoal
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDateTime

@Entity
class DailyMission protected constructor(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mission_id", nullable = false)
    var mission: Mission,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "daily_goal_id", nullable = false)
    var dailyGoal: DailyGoal,
) : BaseTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set

    companion object {
        fun of(
            mission: Mission,
            dailyGoal: DailyGoal,
        ): DailyMission = DailyMission(mission, dailyGoal)
    }

    fun updateDeletedAt(now: LocalDateTime = LocalDateTime.now()) {
        markDeleted(now)
    }
}
